egs/wsj/s5/steps/nnet3/multilingual/allocate_multilingual_examples.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Pegah Ghahremani
+#
+# Apache 2.0.
+
+""" This script generates egs.archive.scp and ranges.* used for multilingual setup.
+    Also this script generates outputs.*.ark and weight.*.ark, where each line
+    corresponds to language-id and weight for the same example in egs.*.scp.
+    weight.*.ark is archive of table indexed by the key of input example
```

Comment:
change 'weight.*.ark is archive of table indexed by the key of input example
 used to scale the eg's output supervision during training.' to:
'weights.*.ark map from the key of the example to the language-specific weight
 of that example.'
You can remove: "    Scaling output supervision is the same as scaling the derivative during
     training for linear objectives."
Change: "   output.*.ark is archive of table indexed by the key of input example and
 +    value of output-name as target language for eg."
to: "output.*.ark map from the key of the example to the name of the output-node
  in the neural net for that specific language, e.g. 'output-2'."

Change:
'ranges.*.scp is randomly generated list of examples from input languages
    generated using frequency distribution of remaining examples in each language.'
to:
```
ranges.*.txt are files, each corresponding to one 'egs' file that will be generated,
where each line corresponds to a range of lines in one of the input languages' files.
The format is:
 <source-language> <start-line> <num-lines>
e.g.
 lang1 1024 256
```
... however, before changing that, can you please tell me how the script that uses this,
maps from e.g. 'lang1' to the source filename?  I hope it doesn't assume a particular
directory structure?
note that I am saying to change the  names of ranges.*.scp to ranges.*.txt, because they are not
scp files (they are not read by any Kaldi binary).
I can't figure out where ranges.*.txt are consumed.  Please document that.